roomserver/state/state.go

```diff
@@ -681,6 +681,60 @@ func (v StateResolution) calculateStateAfterManyEvents(
 	return
 }
 
+// ResolveConflictsAdhoc is a helper function to assist the query API in
+// performing state resolution when requested. This is a different code
+// path to the rest of state.go because this assumes you already have
+// gomatrixserverlib.Event objects and not just a bunch of NIDs like
+// elsewhere in the state resolution.
+// TODO: Some of this can possibly be deduplicated
+func ResolveConflictsAdhoc(
+	version gomatrixserverlib.RoomVersion,
+	events []gomatrixserverlib.Event,
+	authEvents []gomatrixserverlib.Event,
+) ([]gomatrixserverlib.Event, error) {
```

Comment:
Unclear what the return arg is. `resolved` I guess is a strict subset of `events`? It's guaranteed to only be state events?

Reply:
I've added a comment, but yes, it's the resolved state plus unconflicted state, and they will only be state events.